cpp/src/transform/transform.cpp

```diff
@@ -63,6 +65,74 @@ void unary_operation(mutable_column_view output,
              cudf::jit::get_data_ptr(input));
 }
 
+std::vector<std::string> make_template_types(column_view outcol_view, table_view data_view)
+{
+  std::string mskptr_type =
+    cudf::jit::get_type_name(cudf::data_type(cudf::type_to_id<cudf::bitmask_type>())) + "*";
+  std::string offset_type =
+    cudf::jit::get_type_name(cudf::data_type(cudf::type_to_id<cudf::offset_type>()));
+
+  std::vector<std::string> template_types(1);
+  template_types[0] = cudf::jit::get_type_name(outcol_view.type());
+
+  for (auto const& col : data_view) {
+    auto these_types = {cudf::jit::get_type_name(col.type()) + "*", mskptr_type, offset_type};
+
+    template_types.insert(template_types.end(), these_types);
+  }
+  return template_types;
+}
+
+void generalized_operation(table_view data_view,
+                           std::string const& binary_udf,
+                           data_type output_type,
+                           column_view const& outcol_view,
+                           column_view const& outmsk_view,
+                           rmm::mr::device_memory_resource* mr)
+{
+  std::vector<std::string> template_types = make_template_types(outcol_view, data_view);
+
+  std::string generic_kernel_name =
+    jitify2::reflection::Template("cudf::transformation::jit::generic_udf_kernel")
+      .instantiate(template_types);
+
+  std::string generic_cuda_source = cudf::jit::parse_single_function_ptx(
+    binary_udf, "GENERIC_OP", cudf::jit::get_type_name(output_type), {0});
+
+  // {size, out_ptr, out_mask_ptr, col0_ptr, col0_mask_ptr, col0_offset, col1_ptr...}
+  std::vector<void*> kernel_args((data_view.num_columns() * 3) + 3);
```

Comment:
This section of the code clearly needs work - fingers crossed that someone more knowledgeable in c++ can point me in the right direction here :)

Reply:
Should be updated